SH4RE/SH4RE/Messages/MessagesInboxView.swift

```diff
@@ -0,0 +1,200 @@
+//
+//  MessagesView.swift
+//  SH4RE
+//
+//  Created by Hannah Brooks on 2022-11-24.
+//
+
+import SwiftUI
+import Firebase
+import FirebaseFirestoreSwift
+
+class MainMessagesViewModel: ObservableObject {
+    
+    @Published var errorMessage = ""
+    @Published var chatUser: ChatUser?
+    @Published var isUserCurrentlyLoggedOut = false
+    
+    init() {
+        
+        DispatchQueue.main.async {
+            self.isUserCurrentlyLoggedOut = FirebaseManager.shared.auth.currentUser?.uid == nil
+        }
+        
+        fetchCurrentUser()
+        fetchRecentMessages()
+    }
+    
+    
+    @Published var recentMessages = [RecentMessage]()
+    
+    private var firestoreListener: ListenerRegistration?
+    
+    func fetchRecentMessages() {
+        guard let uid = FirebaseManager.shared.auth.currentUser?.uid else { return }
```

Comment:
theres a error msg below if "else" but not here, is there a reason?

Reply:
question for @bbryanb333

Reply:
This should never really be an error but I added the guard statement so we could properly cast later on